src/Microsoft.ML.Data/Prediction/Calibrator.cs

```diff
@@ -186,27 +209,31 @@ void ICanSaveSummary.SaveSummary(TextWriter writer, RoleMappedSchema schema)
 
         private protected void SaveCore(ModelSaveContext ctx)
         {
-            ctx.SaveModel(SubPredictor, ModelFileUtils.DirPredictor);
+            ctx.SaveModel(SubModelParameters, ModelFileUtils.DirPredictor);
             ctx.SaveModel(Calibrator, @"Calibrator");
         }
 
-        private protected static IPredictorProducing<float> GetPredictor(IHostEnvironment env, ModelLoadContext ctx)
+        private protected static TSubModelParameters GetPredictor(IHostEnvironment env, ModelLoadContext ctx)
         {
-            IPredictorProducing<float> predictor;
-            ctx.LoadModel<IPredictorProducing<float>, SignatureLoadModel>(env, out predictor, ModelFileUtils.DirPredictor);
+            TSubModelParameters predictor;
+            ctx.LoadModel<TSubModelParameters, SignatureLoadModel>(env, out predictor, ModelFileUtils.DirPredictor);
             return predictor;
         }
 
-        private protected static ICalibrator GetCalibrator(IHostEnvironment env, ModelLoadContext ctx)
+        private protected static TCalibrator GetCalibrator(IHostEnvironment env, ModelLoadContext ctx)
         {
-            ICalibrator calibrator;
-            ctx.LoadModel<ICalibrator, SignatureLoadModel>(env, out calibrator, @"Calibrator");
+            TCalibrator calibrator;
+            ctx.LoadModel<TCalibrator, SignatureLoadModel>(env, out calibrator, @"Calibrator");
             return calibrator;
         }
     }
 
-    public abstract class ValueMapperCalibratedPredictorBase : CalibratedPredictorBase, IValueMapperDist, IFeatureContributionMapper, ICalculateFeatureContribution,
+    internal abstract class ValueMapperCalibratedPredictorBase<TSubPredictor,TCalibrator> :
```

Comment:
>TSubPredictor [](start = 63, length = 13)

Could you rename to `TSubModelParameters`, to be more consistent with the above?

Reply:
Sure.

---
In reply to: [255159935](https://github.com/dotnet/machinelearning/pull/2435#discussion_r255159935) [](ancestors = 255159935)